### PR TITLE
NO-ISSUE: remove deprecated client.update_hosts func

### PR DESCRIPTION
**Pull request by danielerez**
Removed client.update_hosts func and usage in utils.
Should use update_host instead for fixing day2 flow.

### CODE REVIEW

**Review by osherdp — APPROVED** [review]

File: src/assisted_test_infra/test_infra/utils/utils.py [hunk 1]
```diff
@@ -561,7 +561,8 @@ def update_hosts(client, cluster_id, libvirt_nodes, update_hostnames=False, upda
     if not update_roles:
         roles = None
 
-    client.update_hosts(cluster_id=cluster_id, hosts_with_roles=roles, hosts_names=hostnames)
```
Comment (eliorerz) [inline src/assisted_test_infra/test_infra/utils/utils.py:564]: Why not replacing it in this PR? Why split it into 2 PRs?

**Comment by osherdp** [comment]
> /test e2e-metal-assisted-day2

day2 will probably be broken for a short period of time

**Comment by osherdp** [comment]
/hold
(until live-iso job is fixed)

**Review by nmagnezi — COMMENTED** [review]
Rebase this PR on master so https://github.com/openshift/assisted-test-infra/pull/1529 gets included in your branch. That might help with the live-iso job issue.

File: src/assisted_test_infra/test_infra/utils/utils.py (hunk 1 above)
Comment (danielerez) [reply src/assisted_test_infra/test_infra/utils/utils.py:564]: It just needed more work and testings. Handled in this PR: https://github.com/openshift/assisted-test-infra/pull/1525

**Review by osherdp — APPROVED** [review]

**Pull request merged**